app/models/course.rb

```diff
@@ -142,6 +141,13 @@ def students
     course_user_data.where(course_assistant: false, instructor: false, dropped: [false, nil])
   end
 
+  def assessment_categories
+    assessments.pluck(:category_name).uniq.sort
+  end
+
+  def assessments_with_category(cat_name)
```

Comment:
I think assessments_in_category or assessments_by_category might be more descriptive names, but it's your call.